packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/utils.ts

```diff
@@ -0,0 +1,17 @@
+import isEqual from "lodash/isEqual";
+import { IdentityJoinQuery } from "back-end/types/datasource";
+
+/**
+ * Given a new list of IDs, checks if they're in the existing list of identity joins
+ * @param newIds
+ * @param existingIdentityJoins
+ */
+export const isDuplicateIdentityJoin = (
+  newIds: string[],
+  existingIdentityJoins: IdentityJoinQuery[]
+): boolean => {
+  const existingElement = existingIdentityJoins.find((item) => {
+    return isEqual(item.ids.sort(), newIds.sort());
```

Comment:
`.sort()` mutates the original array.  Not sure we want that side effect in a helper function.

Reply:
Good call. I pushed up a change that uses lodash's `sortBy` which should avoid mutating the arrays.